### PR TITLE
queries for entities (artist, user)

### PR DESCRIPTION
**Pull request by vansika**
Made changes in artist.py

### CODE REVIEW

File: listenbrainz_spark/stats/artist.py [hunk 1]
```diff
@@ -2,27 +2,105 @@
 import time
 from listenbrainz_spark import config
 from listenbrainz_spark.stats import run_query
+from dateutil.relativedelta import relativedelta
+from datetime import datetime 
+from collections import defaultdict
 
+def get_releases():
+    result = run_query("""
```
Comment (paramsingh) [inline listenbrainz_spark/stats/artist.py]: We need to order this by the most listened to release for each artist.

File: listenbrainz_spark/stats/artist.py [hunk 2]
```diff
@@ -2,27 +2,105 @@
...
+    artist_count = defaultdict(list)
+    for row in result.collect():
+        artist_count[row.artist_msid].append({
+            'artist_count' : row.cnt,
+        })
+    return artist_count
 
-
+def get_artist_names():
+    result = run_query("""
```
Comment (paramsingh) [inline listenbrainz_spark/stats/artist.py]: This doesn't seem to calculate any stats.

File: listenbrainz_spark/stats/artist.py [hunk 3]
```diff
@@ -2,27 +2,105 @@
...
+    for row in result.collect():
+        listener[row.artist_msid].append({
+            'user_name': row.user_name,
+            'listen_count' : row.cnt,
+        })
+    return listener
 
+def get_listen_count():
+    result = run_query("""
+        SELECT artist_msid
```
Comment (paramsingh) [inline listenbrainz_spark/stats/artist.py]: We should get more details here, such as the artist name.

File: listenbrainz_spark/stats/artist.py [hunk 4]
```diff
@@ -2,9 +2,73 @@
...
+def get_listener():
+    result = run_query("""
+        SELECT artist_msid
+             , user_name
+             , count(artist_msid) as cnt
+          FROM listen
+      GROUP BY artist_msid, user_name
+      ORDER BY cnt DESC
+    """)
+    result.show()
```
Comment (paramsingh) [inline listenbrainz_spark/stats/artist.py]: Lets remove these shows

File: listenbrainz_spark/stats/artist.py [hunk 5]
```diff
@@ -2,9 +2,73 @@
...
+        recordings[row.artist_msid].append({
+            'track_name' : row.track_name,
+            'recording_msid' : row.recording_msid,
+            'recording_mbid' : row.recording_mbid,
+            'cnt' : row.cnt,
+        })
+    return recordings
 
-def get_most_popular():
+def get_listener():
```
Comment (paramsingh) [inline listenbrainz_spark/stats/artist.py]: Docstrings explaining what each function does would be helpful

File: listenbrainz_spark/stats/artist.py [hunk 6]
```diff
@@ -2,9 +2,73 @@
...
+    return releases
+
+def get_recordings():
+    result = run_query("""
+        SELECT artist_msid
+             , track_name
+             , recording_mbid
+             , recording_msid
+             , count(recording_msid) as cnt
+          FROM listen
```
Comment (paramsingh) [inline listenbrainz_spark/stats/artist.py]: In the other PR, we pass around the name of the temporary table. I know I am the one who started the name listen but we should have better names and pass them around too (in consistency with the other parts of the codebase)

File: listenbrainz_spark/stats/release.py [hunk 7]
```diff
@@ -2,29 +2,76 @@
...
+        }
+    return release_count
 
 def main(app_name):
     t0 = time.time()
     listenbrainz_spark.init_spark_session(app_name)
     df = None
-    for y in range(2018, 2019):
-        for m in range(12, 13):
+    t = datetime.utcnow().replace(day=1)
```
Comment (paramsingh) [inline listenbrainz_spark/stats/release.py]: This code is duplicated a lot, we should consider creating a function that takes a time range and returns a dataframe with listens between that timeframe.

Comment (vansika) [reply listenbrainz_spark/stats/release.py]: The loop was for testing purpose. Like we have decided, stats would be calculated for the previous month/week, this code would no longer be needed.

**Comment by paramsingh** [comment]
Merging this into the `popular-artist` branch and gonna continue from there.

**Pull request merged**